test/library/stark/test_ramsey_xy.py

```diff
@@ -0,0 +1,125 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Stark Ramsey XY experiments."""
+
+from test.base import QiskitExperimentsTestCase
+
+import numpy as np
+from qiskit import pulse
+from qiskit.providers.fake_provider import FakeHanoiV2
+
+from qiskit_experiments.library import StarkRamseyXY
+
+
+class TestStarkRamseyXY(QiskitExperimentsTestCase):
+    """Test case for the Stark Ramsey XY experiment.
+
+    Because the analysis is identical to the RamseyXY, integration test and
+    test for analysis class is skipped here. These must be covered by
+    test/library/calibration/test_ramsey_xy.py
+    """
+
+    def test_calibration_with_positive_amp(self):
+        """Test Stark frequency shift chooses the proper sign with positive amplitude."""
+        backend = FakeHanoiV2()
+        exp = StarkRamseyXY(
+            qubit=0,
+            stark_amp=0.1,  # positive amplitude
+            backend=backend,
+            stark_sigma=15e-9,
+            stark_risefall=2,
+            stark_freq_offset=80e6,
+        )
+        param_ram_x, _ = exp.parameterized_circuits()
+        freq = backend.qubit_properties(0).frequency - 80e6  # negative frequency shift
+        granularity = backend.target.granularity
+        dt = backend.dt
+        duration = granularity * int(round(4 * 15e-9 / dt / granularity))
+        sigma = duration / 4
+
+        with pulse.build() as ref_schedule:
+            pulse.set_frequency(freq, pulse.DriveChannel(0))
+            pulse.play(
+                pulse.Gaussian(duration=duration, amp=0.1, sigma=sigma), pulse.DriveChannel(0)
+            )
+
+        test_schedule = param_ram_x.calibrations["StarkV"][(0,), ()]
+        self.assertEqual(test_schedule, ref_schedule)
```

Comment:
This is fine, but I wonder if it would be cleaner to check that `test_schedule` starts with the right `set_frequency` instruction instead of comparing the full schedules?

Reply:
I also wanted to check if amp if always-positive so I needed to check full schedule. I also added the comment to tests.

Reply:
Ah, okay, though a negative amp shouldn't hurt anything :slightly_smiling_face:

Reply:
Yes, exactly. At the same time I also found a cubit term was sometimes introduced at high power, so this prevents user from using unexpected frequency. And this is a strict form the amp defined in the manual.

Reply:
Hmm, what prevents the user from using an unexpected frequency? I didn't follow the connection between the amp sign and the cubic term.